flink-runtime-web/web-dashboard/src/app/pages/task-manager/metrics/task-manager-metrics.component.html

```diff
@@ -294,3 +294,75 @@
     </div>
   </div>
 </nz-card>
+
+<ng-template #tmResourceRow let-detail="detail" let-name="name" let-property="prop">
+  <tr>
+    <td>{{ name }}</td>
+    <td>
+      <nz-progress
+        nzSize="small"
+        nzStrokeLinecap="square"
+        [nzPercent]="
+          +(
+            (detail?.freeResource[property] / detail?.totalResource[property]) * 100
+            | number: '1.0-2'
+          )
+        "
+        nzStatus="normal"
+      ></nz-progress>
+      {{ detail?.freeResource[property] | number }} / {{ detail?.totalResource[property] | number }}
```

Comment:
consider adding units (especially for memory-related rows)
```suggestion
      {{ detail?.freeResource[property] | number }} / {{ detail?.totalResource[property] | number }} {{ unit }}
```